diff --git a/pypy/translator/platform/posix.py b/pypy/translator/platform/posix.py
--- a/pypy/translator/platform/posix.py
+++ b/pypy/translator/platform/posix.py
@@ -48,8 +48,10 @@
         response_file = self._make_response_file("dynamic-symbols-")
         f = response_file.open("w")
         f.write("{\n")
+        f.write("  global:\n")
         for sym in eci.export_symbols:
-            f.write("%s;\n" % (sym,))
+            f.write("    %s;\n" % (sym,))
+        f.write("  local:*;\n")
         f.write("};")
         f.close()
 
