--- pypy-pypy-release-1.7/pypy/module/_minimal_curses/fficurses.py~	2011-11-19 08:44:54.000000000 +0100
+++ pypy-pypy-release-1.7/pypy/module/_minimal_curses/fficurses.py	2012-02-02 14:10:59.067360229 +0100
@@ -10,7 +10,7 @@
 from pypy.translator.tool.cbuild import ExternalCompilationInfo
 
 eci = ExternalCompilationInfo(
-    includes = ['curses.h', 'term.h'],
+    includes = ['ncurses/curses.h', 'ncurses/term.h'],
     libraries = ['curses'],
 )
 
