--- ./pypy/module/_minimal_curses/fficurses.py.curses	2012-06-07 14:24:48.000000000 +0200
+++ ./pypy/module/_minimal_curses/fficurses.py	2012-08-03 11:54:28.717067645 +0200
@@ -9,10 +9,12 @@ from pypy.rpython.extfunc import registe
 from pypy.module._minimal_curses import interp_curses
 from pypy.translator.tool.cbuild import ExternalCompilationInfo
 from sys import platform
+import os.path
 
 _CYGWIN = platform == 'cygwin'
+_NCURSES_CURSES = os.path.isfile("/usr/include/ncurses/curses.h") 
 
-if _CYGWIN:
+if _CYGWIN or _NCURSES_CURSES:
     eci = ExternalCompilationInfo(
         includes = ['ncurses/curses.h', 'ncurses/term.h'],
         libraries = ['curses'],
